src/Cake.Svn/SvnAliases.Add.cs

```diff
@@ -1,51 +1,113 @@
         /// <example>
-        /// <para>Add a file or a directory to Subversion.</para>
+        /// <para>Add a directory to Subversion.</para>
+        /// <code>
+        /// <![CDATA[ 
+        ///     var directoryAdded = SvnAdd(
+        ///         new DirectoryPath(@"C:\project\src\"));
+        ///
+        ///     Verbose("Directory added: {0}", directoryAdded);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Add")]
+        [CakeNamespaceImport("Cake.Svn.Add")]
+        public static bool SvnAdd(this ICakeContext context, DirectoryPath directory)
+        {
+            return SvnAdd(context, directory, null);
+        }
+
+        /// <summary>
+        /// Add a directory to Subversion.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="directory">The directory.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>
+        /// <c>true</c> if the command were successful. Otherwise <c>false</c> will be returned.
+        /// </returns>
+        /// <example>
+        /// <para>Add a directory to Subversion.</para>
         /// <code>
         /// <![CDATA[
-        ///     var fileAdded = SvnAdd(
-        ///         @"C:\project\src\newfile.cs");
+        ///     var svnAddSettings = 
+        ///         new SvnAddSettings
+        ///         {
+        ///         Force = true
+        ///         };
         ///
         ///     var directoryAdded = SvnAdd(
-        ///         @"C:\project\src\newfolder\");
+        ///         new DirectoryPath(@"C:\project\src\"));
         ///
-        ///     Verbose("File added: {0}", fileAdded);
         ///     Verbose("Directory added: {0}", directoryAdded);
         /// ]]>
         /// </code>
         /// </example>
         [CakeMethodAlias]
         [CakeAliasCategory("Add")]
         [CakeNamespaceImport("Cake.Svn.Add")]
-        public static bool SvnAdd(this ICakeContext context, string fileOrDirectoryToAdd)
+        public static bool SvnAdd(this ICakeContext context, DirectoryPath directory, SvnAddSettings settings)
         {
-            var commands = new SvnAdder(context.Environment, SvnClientFactoryMethod);
+            var adder = new SvnAdder(context.Environment, SvnClientFactoryMethod);
 
-            return SvnAdd(context, fileOrDirectoryToAdd, null);
+            return adder.Add(directory, settings ?? new SvnAddSettings());
         }
 
         /// <summary>
-        /// Add a file or directory to Subversion.
+        /// Add a file to Subversion.
         /// </summary>
         /// <param name="context">The Cake context.</param>
-        /// <param name="fileOrDirectoryToAdd">The filename or directoryname to add.</param>
+        /// <param name="file">The file.</param>
+        /// <returns>
+        /// <c>true</c> if the command were successful. Otherwise <c>false</c> will be returned.
```

Comment:
`Returns <c>true</c> if the command was successful. Otherwise <c>false</c> will be returned.`